test/test_types.js

```diff
@@ -2130,6 +2130,52 @@ suite('types', function () {
       var E = t.getRecordConstructor();
       var err = new E('MyError');
       assert(err instanceof Error);
+      if (typeof Error.captureStackTrace === 'function') {
+        assert(err.stack, 'Error.captureStackTrace was not called');
+      }
+    });
+
+    test('error type - stack is not overridden', function() {
+
+      // This test only applies to V8 environments
+      if (typeof Error.captureStackTrace !== 'function') {
+        return;
+      }
+
+      var t = Type.forSchema({
+        type: 'error',
+        name: 'Ouch',
+        fields: [
+          {name: 'name', type: 'string'},
+          {name: 'stack', type: 'string'},
+        ]
+      });
+
+      var E = t.getRecordConstructor();
+      var err = new E('MyError', 'my amazing stack');
+      assert(err instanceof Error);
+      assert(err.stack === 'my amazing stack');
```

Comment:
Can you add a check for when no stack value is passed in as argument (ensuring that the stack is populated)?